docs/rules/no-tracked-properties-from-args.md

```diff
@@ -0,0 +1,29 @@
+# ember/no-tracked-properties-from-args
+
+<!-- end auto-generated rule header -->
+
+Disallow creation of @tracked properties from args.
+
+## Rule Details
+
+This rule disallows the creation of @tracked properties with values from `this.args`. The @tracked property will not be updated when the args change, which is almost never what you want.
+
+## Examples
+
+Examples of **incorrect** code for this rule:
+
+```js
+@tracked someValue = this.args.someValue;
+```
+
+Examples of **correct** code for this rule:
+```js
+get someValue() {
+    return this.args.someValue;
+}
+```
```

Comment:
Does adding a getter for the arg actually address the user's original intent in using `@tracked`? Does it mean that `@tracked` was entirely unnecessary?

Reply:
I would still like to understand this a bit better, especially as to whether a getter is really a complete fix for what the user is attempting to achieve? We don't have to show a complete fix necessarily, but I would like to make sure we describe at a high-level what kind of more-involved fixes might be necessary.

Reply:
For the simple use case of trying to rename a deeply nested arg, @tracked would be unnecessary and it would just be used interchangeably with a getter. Trying to raise awareness about the extra pitfall here.
I added a note on having to mutate the args to the docs portion, let me know if that helps!